client/ayon_core/modules/deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -490,15 +490,19 @@ def inject_ayon_environment(deadlinePlugin):
         if job.GetJobEnvironmentKeyValue("IS_TEST"):
             args.append("--automatic-tests")
 
-        if all(add_kwargs.values()):
-            for key, value in add_kwargs.items():
-                args.extend(["--{}".format(key), value])
-        else:
+        if not all(add_kwargs.values()):
             raise RuntimeError((
                 "Missing required env vars: AYON_PROJECT_NAME,"
                 " AYON_FOLDER_PATH, AYON_TASK_NAME, AYON_APP_NAME"
             ))
 
+        legacy_args = list(args)
+        for key, value in add_kwargs.items():
+            args.extend(["--{}".format(key), value])
+            if key == "folder":
+                key = "asset"
+            legacy_args.extend(["--{}".format(key), value])
```

Comment:
Should this logic be inside the first failure since it's only used in there anyway?

Reply:
You mean recalculate all arguments again?

Reply:
Ah sorry - I missed that this was _also_ setting the regular `args`. Nevermind me.